components/support/nimbus-fml/src/backends/swift/templates/FeatureManifestTemplate.swift

```diff
@@ -36,6 +36,14 @@ public class {{ nimbus_object }} {
     ///
     public let features = {{ nimbus_object }}Features()
 
+    /// Clear all cached values for all features.
+    /// This should be called after `applyPendingExperiments` is finished.
+    public func invalidateCachedValues() {
```

Comment:
Just for my own clarity, I assume this is needed to be called after `applyPendingExperiments` is finished because it's possible a `value` was called before initialization, returning a default and we want to wipe that from the cache?

This makes me think a little about the "Nimbus is Idompotent" contract we have with mobile and whether or not this violates that

I may be missing a part of this, so maybe we can add some more context somewhere (not sure if this fits here in the doc comment, but may be the most convenient?

Reply:
Yes, this is correct. If we don't call it after `applyPendingExperiments` finishes then a call before the end of initialization (which gets less and less likely now) will poison the well for every subsequent call.

The idempotency illusion really is and always has been "idempotent between calls to `applyPendingExperiments`). This cache invalidation is needed to maintain that illusion.